src/common/hooks/useSafeMemo.ts

```diff
@@ -1,15 +1,16 @@
-import { useMemo } from 'react'
+import { EffectCallback, useEffect, useMemo } from 'react'
 
-import { CurrencyAmount, NativeCurrency, Token } from '@uniswap/sdk-core'
+import { CurrencyAmount, NativeCurrency, Percent, Token } from '@uniswap/sdk-core'
 
 import { WrappedTokenInfo } from 'legacy/state/lists/wrappedTokenInfo'
 
-export function useSafeMemoDeps(deps: unknown[]): unknown[] {
+export function useSafeDeps(deps: unknown[]): unknown[] {
   return deps.map((dep) => {
     if (dep instanceof NativeCurrency) return dep.symbol
     if (dep instanceof Token) return dep.address.toLowerCase()
     if (dep instanceof CurrencyAmount) return dep.toExact() + dep.currency.symbol + dep.currency.isNative
     if (dep instanceof WrappedTokenInfo) return dep.address
+    if (dep instanceof Percent) return dep.toFixed(6)
```

Comment:
Is this enough to check it's different, right?

Actually, now that I'm seeing this I'm considering making it generic for any fraction with https://github.com/cowprotocol/cowswap/blob/0ae6602f707045079c8a5e0f3b8d6bd70f703e61/src/utils/fractionUtils.ts#L12-L11